sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/filter/UserPrincipalManager.java

```diff
@@ -151,7 +150,7 @@ public UserPrincipal buildUserPrincipal(String aadIssuedBearerToken) throws Pars
         UserPrincipal userPrincipal = new UserPrincipal(aadIssuedBearerToken, jwsObject, jwtClaimsSet);
         Set<String> roles = Optional.of(userPrincipal)
                                     .map(p -> p.getClaim(AadJwtClaimNames.ROLES))
-                                    .map(r -> (JSONArray) r)
```

Comment:
Fix issue: Replace this lambda with method reference 'JSONArray.class::cast'.

refer: https://sonarcloud.io/project/issues?directories=sdk%2Fspring%2Fspring-cloud-azure-autoconfigure%2Fsrc%2Fmain%2Fjava%2Fcom%2Fazure%2Fspring%2Fcloud%2Fautoconfigure%2Faad%2Ffilter&resolved=false&types=CODE_SMELL&id=stliu_azure-sdk-for-java&open=AXtAG3Fj00FAjxsg1BZP